runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -7642,6 +7643,17 @@ done:;
 				goto retry;
 			}
 		}
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+		else {
+			if ((NULL != castClass) && J9_IS_J9CLASS_VALUETYPE(castClass)) {
+				rc = THROW_NPE;
```

Comment:
Is NPE the correct result here?

Reply:
Yes according to the <a href="http://cr.openjdk.java.net/~dlsmith/lw2/lw2-20190628/specs/inline-classes-jvms.html#jvms-6.5.checkcast">lw2 spec</a> `if objectref is null and the symbolic reference names a null-free inline class type, the checkcast instruction throws a NullPointerException.`